kernel/qsort.c

```diff
@@ -0,0 +1,266 @@
+/*
```

Comment:
An alternative of copying newlib's qsort.c here would be to use the one in /lib/newlib/newlib/libc/search/qsort.c directly. I'm undecided what makes more sense.